driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java

```diff
@@ -220,15 +222,11 @@ public Session getSession() {
     private <T> Mapper<T> getMapper(Class<T> klass) {
         Mapper<T> mapper = (Mapper<T>) mappers.get(klass);
         if (mapper == null) {
-            synchronized (mappers) {
-                mapper = (Mapper<T>) mappers.get(klass);
-                if (mapper == null) {
-                    EntityMapper<T> entityMapper = AnnotationParser.parseEntity(klass, this);
-                    mapper = new Mapper<T>(this, klass, entityMapper);
-                    Map<Class<?>, Mapper<?>> newMappers = new HashMap<Class<?>, Mapper<?>>(mappers);
-                    newMappers.put(klass, mapper);
-                    mappers = newMappers;
-                }
+            EntityMapper<T> entityMapper = AnnotationParser.parseEntity(klass, this);
+            mapper = new Mapper<T>(this, klass, entityMapper);
+            Mapper<T> old = (Mapper<T>) mappers.putIfAbsent(klass, mapper);
+            if (old != null) {
+                mapper = old;
```

Comment:
This is not exactly the same as the previous version, now we create a new instance each time, even if the entry existed in the map.
For `getMapper` and `getAccessor` it doesn't matter that much, because they're only called directly by the client (and in general they will probably only do it once), but `getUDTCodec` is also called internally, each time a new class with an UDT field is parsed.
Checking the map first is simple enough, so I think we should do it in all cases:
```
Mapper<T> old = (Mapper<T>) mappers.get(klass);
if (old != null) {
    return old;
}
EntityMapper<T> entityMapper = AnnotationParser.parseEntity(klass, this);
mapper = new Mapper<T>(this, klass, entityMapper);
old = (Mapper<T>) mappers.putIfAbsent(klass, mapper);
if (old != null) {
    mapper = old;
}
return mapper;
```
It can still create an unnecessary instance, but the window is much shorter.